DataFormats/Detectors/TPC/src/PIDResponse.cxx

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// @file   PIDResponse.cxx
+/// @author Tuba Gündem, tuba.gundem@cern.ch
+///
+
+#include "DataFormatsTPC/PIDResponse.h"
+#include "DataFormatsTPC/BetheBlochAleph.h"
+
+using namespace o2::tpc;
+
+double PIDResponse::getExpectedSignal(const TrackTPC& track, const o2::track::PID::ID id) const
+{
+  const double bethe = mMIP * o2::tpc::BetheBlochAleph(static_cast<double>(track.getP() / o2::track::pid_constants::sMasses[id]), mBetheBlochParams[0], mBetheBlochParams[1], mBetheBlochParams[2], mBetheBlochParams[3], mBetheBlochParams[4]) * std::pow(static_cast<float>(o2::track::pid_constants::sCharges[id]), mChargeFactor);
+  return bethe >= 0.f ? bethe : -999.f;
```

Comment:
Didn't we discuss to put some lower bg value (could also be made configurable)? Also, your function is expected to return double.
Something like:
```suggestion
  const double bg = static_cast<double>(track.getP() / o2::track::pid_constants::sMasses[id]);
  if (bg < 0.05) {
    return -999.;
  }
  const double bethe = mMIP * o2::tpc::BetheBlochAleph(bg, mBetheBlochParams[0], mBetheBlochParams[1], mBetheBlochParams[2], mBetheBlochParams[3], mBetheBlochParams[4]) * std::pow(static_cast<float>(o2::track::pid_constants::sCharges[id]), mChargeFactor);
  return bethe >= 0. ? bethe : -999.;
```